src/vanilla/utils/atomWithStorage.ts

```diff
@@ -99,8 +99,18 @@ export function createJSONStorage<Value>(
   getStringStorage: () =>
     | AsyncStringStorage
     | SyncStringStorage
-    | undefined = () =>
-    typeof window !== 'undefined' ? window.localStorage : undefined,
+    | undefined = () => {
+    try {
+      return window.localStorage
+    } catch (e) {
+      // Window could be undefined or local storage could be disabled
+      if (e instanceof TypeError || e instanceof DOMException) {
```

Comment:
Hmm, this is what I tried in Node.js:
```
> try { window.localStorage } catch (e) { console.log(e instanceof TypeError) }
false
undefined
> try { window.localStorage } catch (e) { console.log(e instanceof ReferenceError) }
true
undefined
```
Maybe, it's not stable. On the other hand, when is the case we want to `throw e`?

Reply:
Oh, it seems on node the window instance is not even defined.
I'm wondering under which circumstances the window instance is not defined, but this library is used?
As far as I understand, Jotai only works alongside React?

Reply:
SSR=Server Side Rendering.